python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -675,39 +675,21 @@ def drain_nodes_via_gcs(self, provider_node_ids_to_drain: List[NodeID]):
 
         logger.info(f"Draining {len(raylet_ids_to_drain)} raylet(s).")
         try:
-            request = gcs_service_pb2.DrainNodeRequest(
-                drain_node_data=[
-                    gcs_service_pb2.DrainNodeData(node_id=raylet_id)
-                    for raylet_id in raylet_ids_to_drain
-                ]
-            )
-
             # A successful response indicates that the GCS has marked the
             # desired nodes as "drained." The cloud provider can then terminate
             # the nodes without the GCS printing an error.
-            response = self.gcs_node_info_stub.DrainNode(request, timeout=5)
-
             # Check if we succeeded in draining all of the intended nodes by
             # looking at the RPC response.
-            drained_raylet_ids = {
-                status_item.node_id for status_item in response.drain_node_status
-            }
+            drained_raylet_ids = set(self.gcs_client.drain_nodes(
+                raylet_ids_to_drain, timeout=5))
             failed_to_drain = raylet_ids_to_drain - drained_raylet_ids
             if failed_to_drain:
                 self.prom_metrics.drain_node_exceptions.inc()
                 logger.error(f"Failed to drain {len(failed_to_drain)} raylet(s).")
-
-        # If we get a gRPC error with an UNIMPLEMENTED code, fail silently.
-        # This error indicates that the GCS is using Ray version < 1.8.0,
-        # for which DrainNode is not implemented.
-        except grpc.RpcError as e:
-            # If the code is UNIMPLEMENTED, pass.
-            if e.code() == grpc.StatusCode.UNIMPLEMENTED:
-                pass
-            # Otherwise, it's a plane old gRPC error and we should log it.
-            else:
-                self.prom_metrics.drain_node_exceptions.inc()
-                logger.exception("Failed to drain Ray nodes. Traceback follows.")
+        except RpcError:
+            # Otherwise, it's a plane old RPC error and we should log it.
```

Comment:
No need to handle `UNIMPLEMENTED` here?

Reply:
The document mentioned it used to deal with Ray version < 1.8.0, and now it's 2.7.0 so I guess we would not ever handle the UNIMPLEMENTED case?